Y/Yosys/build_tarballs.jl

```diff
@@ -3,60 +3,68 @@
 using BinaryBuilder, Pkg
 
 name = "Yosys"
-version = v"0.11.0"
+version = v"0.18.0"
 
 # Collection of sources required to complete build
 sources = [
-    GitSource("https://github.com/YosysHQ/yosys.git", "360fed8e4d611fa725a4526cf960383b8e6c6e64")
+    GitSource("https://github.com/YosysHQ/yosys.git", "19ce3b45d6a877e96af3883e3f8df3acfd224ea6")
 ]
 
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter!(p -> arch(p) != "armv6l", supported_platforms())
+platforms = expand_cxxstring_abis(platforms)
+# For some reasons, building for CXX03 string ABI doesn't actually work, skip it
+filter!(x -> cxxstring_abi(x) != "cxx03", platforms)
+
 dependencies = [
     Dependency("boost_jll"; compat="=1.76.0"), # max gcc7
     Dependency("Readline_jll"; compat="8.1.1"),
     Dependency("Tcl_jll"; compat="8.6.11"),
     Dependency("Zlib_jll"; compat="1.2.11"),
-    Dependency("Libffi_jll"; compat="~3.2.2")
+    Dependency("Libffi_jll"; compat="~3.2.2"),
+    Dependency("dlfcn_win32_jll"; compat="1.3.1", platforms=filter(Sys.iswindows, platforms))
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd yosys
-if [[ "${target}" == *-apple-* ]] || [[ "${target}" == *-freebsd* ]]; then
+
+if [[ "${target}" == *-apple-* ]]; then
     CONFIG=clang
     OS=Darwin
-elif [[ "${target}" == *-x86_64-w64-mingw32* ]]; then
+elif [[ "${target}" == *-freebsd* ]]; then
+    CONFIG=clang
+    OS=FreeBSD
+elif [[ "${target}" == *x86_64-w64-mingw32* ]]; then
     CONFIG=msys2-64
     OS=Windows
-elif [[ "${target}" == *-i686-w64-mingw32* ]]; then
+elif [[ "${target}" == *i686-w64-mingw32* ]]; then
     CONFIG=msys2-32
     OS=Windows
 else
     CONFIG=gcc
     OS=Linux
 fi
-# TODO: we need mingw-dlfcn wrappers for windows plugin support
+
 # TODO: ABC does not build on windows
-make ENABLE_LIBYOSYS=1 OS=${OS} CONFIG=${CONFIG} PREFIX=${prefix} LIBDIR=${libdir} -j${nproc}
-make install ENABLE_LIBYOSYS=1 OS=${OS} PREFIX=${prefix} LIBDIR=${libdir}
+if [[ "${target}" == *x86_64-w64-mingw32* ]] || [[ "${target}" == *i686-w64-mingw32* ]]; then
+    make install ENABLE_LIBYOSYS=1 ENABLE_ABC=0 ENABLE_TCL=0 OS=${OS} CONFIG=${CONFIG} PREFIX=${prefix} TCL_INCLUDE=${libdir} TCL_VERSION=tcl86 LIBDIR=${libdir} -j${nproc}
+else
+    make install ENABLE_LIBYOSYS=1 OS=${OS} CONFIG=${CONFIG} PREFIX=${prefix} TCL_INCLUDE=${libdir} LIBDIR=${libdir} -j${nproc}
+fi
+
 if [[ "${target}" == *-apple-* ]]; then
     mv ${prefix}/lib/libyosys.so  ${prefix}/lib/libyosys.dylib
+elif [[ "${target}" == *-w64-* ]]; then
+    mv ${libdir}/libyosys.so  ${libdir}/libyosys.dll
```

Comment:
This is not what I said in https://github.com/JuliaPackaging/Yggdrasil/pull/5014#discussion_r894936678 :slightly_smiling_face:

Reply:
I tested this locally. I only saw your comment after :)

Reply:
TIL about `${dlext}` :D

Reply:
https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables